app/src/main/java/org/mifos/mobile/ui/fragments/LoanApplicationFragment.kt

```diff
@@ -508,6 +509,21 @@ class LoanApplicationFragment : BaseFragment(), LoanApplicationMvpView, OnDatePi
         loanApplicationPresenter?.detachView()
     }
 
+    private fun purposeListLoader(position: Int) {
+        if (loanTemplate?.loanPurposeOptions?.size == 0) {
+            loanApplicationPresenter?.loadLoanApplicationTemplateByProduct(productId,
+                    LoanState.UPDATE)
+            val handler = Handler()
```

Comment:
Why are we posting delays and making the recursive call? Can we come up with a better solution or can you explain why you came up with this fix ?

Reply:
I wasn't able to find other solution for a long time, but i noticed that loading list of purpose options takes some time so i used handler and recursion is used as based on phone's capability if low memory phone takes longer that specified it will add some more time. If you are not satisfied with results, i will try to change code but need some guidance, because i tried long time and on every method but came no results.